DigitalLearningSolutions.Data/Services/RegistrationService.cs

```diff
@@ -343,32 +389,11 @@ public void PromoteDelegateToAdmin(AdminRoles adminRoles, int? categoryId, int d
             bool registerJourneyContainsTermsAndConditions
         )
         {
-            try
-            {
-                ValidateRegistrationEmail(delegateRegistrationModel);
-                return registrationDataService.RegisterNewUserAndDelegateAccount(
-                    delegateRegistrationModel,
-                    registerJourneyContainsTermsAndConditions
-                );
-            }
-            catch (DelegateCreationFailedException exception)
-            {
-                var error = exception.Error;
-                var errorMessage = $"Could not create account for delegate on registration. Failure: {error.Name}";
-
-                logger.LogError(exception, errorMessage);
-
-                throw new DelegateCreationFailedException(errorMessage, exception, error);
-            }
-            catch (Exception exception)
-            {
-                var error = DelegateCreationError.UnexpectedError;
-                var errorMessage = $"Could not create account for delegate on registration. Failure: {error.Name}";
-
-                logger.LogError(exception, errorMessage);
-
-                throw new DelegateCreationFailedException(errorMessage, exception, error);
-            }
+            ValidateRegistrationEmail(delegateRegistrationModel);
+            return registrationDataService.RegisterNewUserAndDelegateAccount(
+                delegateRegistrationModel,
+                registerJourneyContainsTermsAndConditions
+            );

```

Comment:
How come we're removing the try catch here, What about other places that use this that don't have the error handling/logging? This is also used in the first time registration and the delegate upload journey.

Reply:
Whoops. The try catch was not supposed to be removed from this method. A near identical try catch was supposed to be removed from the private method below it, but I must have got them mixed up when doing the replacement.

The actual try catch that was supposed to be moved (and now should be) simplifies the two private methods for registering/reregistering at a centre for an existing user to prevent a duplication of try catches. As the private methods are only used once, and are within the moved try catch, this should now be fine.

Good spot.